Upstream-Status: Pending

diff --git a/src/main.conf b/src/main.conf
index 4c53376..236468c 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -124,7 +124,7 @@
 
 # Enables D-Bus experimental interfaces
 # Possible values: true or false
-#Experimental = false
+Experimental = true
 
 # Enables D-Bus testing interfaces
 # Possible values: true or false
@@ -141,7 +141,7 @@
 # a6695ace-ee7f-4fb9-881a-5fac66c629af (BlueZ Experimental Offload Codecs)
 # 6fbaf188-05e0-496a-9885-d6ddfdb4e03e (BlueZ Experimental ISO socket)
 # Defaults to false.
-#KernelExperimental = false
+KernelExperimental = 6fbaf188-05e0-496a-9885-d6ddfdb4e03e
 
 # The duration to avoid retrying to resolve a peer's name, if the previous
 # try failed.
