From ee759f40c0db4164f69589126f0126d6fb6cfabc Mon Sep 17 00:00:00 2001
From: Jacob Shen <qi_shen@realsil.com.cn>
Date: Sun, 13 Oct 2024 19:43:56 +0800
Subject: [PATCH] add amebasmart yocto support

Upstream-Status: Pending
---
 src/daemon/pipewire.conf.in | 138 +++++++++++++++++++++++-------------
 1 file changed, 89 insertions(+), 49 deletions(-)

diff --git a/src/daemon/pipewire.conf.in b/src/daemon/pipewire.conf.in
index c3eb7120f..d96bea4a5 100644
--- a/src/daemon/pipewire.conf.in
+++ b/src/daemon/pipewire.conf.in
@@ -10,16 +10,16 @@
 
 context.properties = {
     ## Configure properties in the system.
-    #library.name.system                   = support/libspa-support
-    #context.data-loop.library.name.system = support/libspa-support
-    #support.dbus                          = true
+    library.name.system                   = support/libspa-support
+    context.data-loop.library.name.system = support/libspa-support
+    support.dbus                          = true
     #link.max-buffers                      = 64
     link.max-buffers                       = 16                       # version < 3 clients can't handle more
     #mem.warn-mlock                        = false
     #mem.allow-mlock                       = true
     #mem.mlock-all                         = false
     #clock.power-of-two-quantum            = true
-    #log.level                             = 2
+    log.level                             = 2
     #cpu.zero.denormals                    = false
     #rlimit.nofile                         = -1
 
@@ -44,7 +44,7 @@ context.properties = {
     #default.clock.rate          = 48000
     #default.clock.allowed-rates = [ 48000 ]
     #default.clock.quantum       = 1024
-    #default.clock.min-quantum   = 32
+    default.clock.min-quantum   = 16
     #default.clock.max-quantum   = 2048
     #default.clock.quantum-limit = 8192
     #default.clock.quantum-floor = 4
@@ -76,7 +76,7 @@ context.spa-libs = {
     # that factory.
     #
     audio.convert.* = audioconvert/libspa-audioconvert
-    avb.*           = avb/libspa-avb
+    #avb.*           = avb/libspa-avb
     api.alsa.*      = alsa/libspa-alsa
     api.v4l2.*      = v4l2/libspa-v4l2
     api.libcamera.* = libcamera/libspa-libcamera
@@ -112,7 +112,7 @@ context.modules = [
     { name = libpipewire-module-rt
         args = {
             nice.level    = -11
-            rt.prio       = @rtprio_server@
+            #rt.prio       = 88
             #rt.time.soft = -1
             #rt.time.hard = -1
             #uclamp.min = 0
@@ -184,14 +184,41 @@ context.modules = [
     # new clients.
     { name = libpipewire-module-access
         args = {
-            # Socket-specific access permissions
-            #access.socket = { pipewire-0 = "default", pipewire-0-manager = "unrestricted" }
-
-            # Deprecated legacy mode (not socket-based),
-            # for now enabled by default if access.socket is not specified
-            #access.legacy = true
+            # access.allowed to list an array of paths of allowed
+            # apps.
+            access.allowed = [
+                /usr/bin/wireplumber
+                /usr/bin/wpctl
+                /usr/bin/pw-cat
+                /usr/bin/pw-cli
+                /usr/bin/pw-dot
+                /usr/bin/pw-dsdplay
+                /usr/bin/pw-dump
+                /usr/bin/pw-link
+                /usr/bin/pw-loopback
+                /usr/bin/pw-metadata
+                /usr/bin/pw-mididump
+                /usr/bin/pw-midiplay
+                /usr/bin/pw-midirecord
+                /usr/bin/pw-mon
+                /usr/bin/pw-play
+                /usr/bin/pw-profiler
+                /usr/bin/pw-record
+                /usr/bin/pw-reserve
+                /usr/bin/pw-top
+            ]
+     
+            # An array of rejected paths.
+            #access.rejected = [ ]
+     
+            # An array of paths with restricted access.
+            #access.restricted = [ ]
+     
+            # Anything not in the above lists gets assigned the
+            # access.force permission.
+            #access.force = flatpak
         }
-        condition = [ { module.access = !false } ]
+        #condition = [ { module.access = !false } ]
     }
 
     # Makes a factory for wrapping nodes in an adapter with a
@@ -299,18 +326,52 @@ context.objects = [
         condition = [ { factory.freewheel-driver = !false } ]
     }
 
-    # This creates a new Source node. It will have input ports
-    # that you can link, to provide audio for this source.
-    #{ factory = adapter
-    #    args = {
-    #        factory.name     = support.null-audio-sink
-    #        node.name        = "my-mic"
-    #        node.description = "Microphone"
-    #        media.class      = "Audio/Source/Virtual"
-    #        audio.position   = "FL,FR"
-    #        monitor.passthrough = true
-    #    }
-    #}
+    { factory = adapter
+        args = {
+            factory.name           = api.alsa.pcm.source
+            node.name              = "alsa-source"
+            node.description       = "PCM Source"
+            media.class            = "Audio/Source"
+            api.alsa.path          = "hw:0,1"
+            api.alsa.period-size   = 1024
+            api.alsa.headroom      = 0
+            api.alsa.disable-mmap  = false
+            api.alsa.disable-batch = false
+            audio.format           = "S16LE"
+            audio.rate             = 48000
+            audio.channels         = 2
+            audio.position         = "FL,FR"
+            #priority.driver        = 25000
+        }
+    }
+
+    { factory = adapter
+        args = {
+            factory.name           = "api.alsa.pcm.sink"
+            factory.mode           = "merge"
+            node.name              = "alsa-sink"
+            node.description       = "PCM Sink"
+            media.class            = "Audio/Sink"
+            api.alsa.path          = "hw:0,0"
+            api.alsa.period-size   = 1024
+            api.alsa.headroom      = 0
+            audio.format           = "S16LE"
+            audio.rate             = 48000
+            audio.channels         = 2
+            audio.position         = "FL,FR"
+            device.class           = "sound"
+            #priority.driver        = 25000
+            #node.autoconnect       = true
+            #node.suspend-on-idle   = false
+            #node.pause-on-idle     = false
+            #node.driver            = false
+            #node.passive           = in
+            #api.alsa.disable-mmap  = true
+            #api.alsa.disable-batch = true
+            #resample.disable       = true
+            #channelmix.disable     = true  alsa-driver cannot disable channelmix!
+        }
+    }
 
     # This creates a single PCM source device for the given
     # alsa device path hw:0. You can change source to sink
@@ -346,27 +407,6 @@ context.objects = [
 ]
 
 context.exec = [
-    #{   path = <program-name>
-    #    ( args = "<arguments>" | [ <arg1> <arg2> ... ] )
-    #    ( condition = [ { <key> = <value> ... } ... ] )
-    #}
-    #
-    # Execute the given program with arguments.
-    # If condition is given, the program is executed only when the context
-    # properties all match the match rules.
-    #
-    # You can optionally start the session manager here,
-    # but it is better to start it as a systemd service.
-    # Run the session manager with -h for options.
-    #
-    @sm_comment@{ path = "@session_manager_path@" args = "@session_manager_args@"
-    @sm_comment@  condition = [ { exec.session-manager = !false } ] }
-    #
-    # You can optionally start the pulseaudio-server here as well
-    # but it is better to start it as a systemd service.
-    # It can be interesting to start another daemon here that listens
-    # on another address with the -a option (eg. -a tcp:4713).
-    #
-    @pulse_comment@{ path = "@pipewire_path@" args = [ "-c" "pipewire-pulse.conf" ]
-    @pulse_comment@  condition = [ { exec.pipewire-pulse = !false } ] }
+    # Start Session Manager: WirePlumber
+    { path = "/usr/bin/wireplumber" }
 ]
