From e30d1ccff351aec26dc63414997efb38be8801d0 Mon Sep 17 00:00:00 2001
From: ZivLowRTK <zivlow@realtek-sg.com>
Date: Tue, 27 Aug 2024 17:22:21 +0000
Subject: [PATCH] enable dbus control interface

Upstream-Status: Pending
---
 wpa-supplicant.sh | 2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/wpa-supplicant.sh b/wpa-supplicant.sh
index 35a1aa6..fa451d8 100644
--- a/wpa-supplicant.sh
+++ b/wpa-supplicant.sh
@@ -5,7 +5,7 @@ WPA_SUP_BIN="/usr/sbin/wpa_supplicant"
 WPA_SUP_PNAME="wpa_supplicant"
 WPA_SUP_PIDFILE="/var/run/wpa_supplicant.$IFACE.pid"
 WPA_COMMON_CTRL_IFACE="/var/run/wpa_supplicant"
-WPA_SUP_OPTIONS="-B -P $WPA_SUP_PIDFILE -i $IFACE"
+WPA_SUP_OPTIONS="-B -P $WPA_SUP_PIDFILE -u -i $IFACE"
 
 VERBOSITY=0
