From ff37116ed3fdd0d16623060dbc7547e9bac36b09 Mon Sep 17 00:00:00 2001
From: xuebin_wu <xuebin_wu@realsil.com.cn>
Date: Thu, 10 Jul 2025 14:00:24 +0800
Subject: [PATCH] run rtkcfu before otbr agent start

---
 script/server | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/script/server b/script/server
index 4073936d421..8429f382530 100755
--- a/script/server
+++ b/script/server
@@ -63,6 +63,8 @@ startup()
         sudo service mdns status || sudo service mdns start || echo "service mdns is not available!"
     fi
     without WEB_GUI || start_service otbr-web
+    stop_service otbr-agent
+    sudo rtkcfu update /lib/firmware/rtkcfu/8771HTV/config.txt /lib/firmware/rtkcfu/8771HTV/ /lib/firmware/rtkcfu/8771HTV/ || true
     start_service otbr-agent
 
     # shellcheck source=/dev/null
-- 
2.33.0.windows.1

